.github/workflows/yocto-build-deploy.yml

```diff
@@ -0,0 +1,639 @@
+            -e VERBOSE \
+            -e AWS_ACCESS_KEY_ID \
+            -e AWS_SECRET_ACCESS_KEY \
+            -e AWS_DEFAULT_REGION \
+            -e AWS_SESSION_TOKEN \
+            -e AMI_NAME \
+            -e AMI_ARCHITECTURE \
+            -e AMI_SECUREBOOT \
+            -e S3_BUCKET \
+            -e BALENA_PRELOAD_APP \
+            -e BALENARC_BALENA_URL \
+            -e BALENACLI_TOKEN \
+            -e BALENA_PRELOAD_COMMIT \
+            -e IMAGE \
+            -e MACHINE \
+            -e HOSTOS_VERSION \
+            -e AWS_SUBNET_ID \
+            -e AWS_SECURITY_GROUP_ID \
+            -w "${WORKSPACE}" \
+            "${{ steps.helper-image.outputs.imageid }}" /balena-generate-ami.sh
+
+  balena-deploy:
+    name: HostApp Deploy
+    runs-on: ${{ fromJSON(inputs.runs-on) }}
+    needs: build
+    if: inputs.deploy-hostapp == true
+
+    env:
+      MACHINE: "${{ inputs.machine }}"
+      BALENARC_BALENA_URL: "${{ inputs.balena-environment }}"
+      WORKSPACE: ${{ github.workspace }}
+      VERBOSE: verbose
+      SLUG: "${{ needs.build.outputs.device_slug }}"
+      SECURE_BOOT: "${{ inputs.sign-image }}"
+      SIGN_API_URL: "${{ inputs.sign-api-url }}"
+
+    defaults:
+      run:
+        working-directory: .
+        shell: bash --noprofile --norc -eo pipefail -x {0}
+
+    steps:
+      # TODO: clone balena-yocto-scripts @ yocto_scripts_ref instead
+      # https://github.com/actions/checkout
+      - name: Clone device repository
+        uses: actions/checkout@v3
+        with:
+          repository: ${{ inputs.device-repo || github.repository }}
+          token: ${{ secrets.GITHUB_TOKEN }}
+          ref: ${{ inputs.device-repo-ref || github.ref }}
+          submodules: true
+
+      # https://github.com/actions/download-artifact
+      - name: Download build artifacts
+        uses: actions/download-artifact@eaceaf801fd36c7dee90939fad912460b18a1ffe # v4.1.2
+        with:
+          path: ${{ runner.temp }}
+          name: build-artifacts
+
+      - name: Extract build artifacts
+        run: |
+          tar -xvf ${{ runner.temp }}/build-artifacts.tar.zst
+          find .
+
+      - name: Set deploy outputs
+        run: |
+          if [ -n "${{ inputs.sign-image }}" = "true" ]; then
+            echo "SECURE_BOOT_FEATURE_FLAG=yes" >> $GITHUB_ENV
+          else
+            echo "SECURE_BOOT_FEATURE_FLAG=no" >> $GITHUB_ENV
+          fi
+
+      # https://github.com/docker/setup-buildx-action
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      # https://github.com/docker/build-push-action
+      - name: Build helper image
+        id: helper-image
+        uses: docker/build-push-action@v5.1.0
+        with:
+          context: "${{ github.workspace }}/balena-yocto-scripts/automation"
+          file: Dockerfile_yocto-build-env
+          target: yocto-build-env
+          push: false
+
+      - name: Deploy to balena
```

Comment:
Here is were the deploy to balena action would come in.

Reply:
Correct, will need a TODO here as well.